libpod/runtime_ctr.go

```diff
@@ -171,7 +171,7 @@ func (r *Runtime) newContainer(ctx context.Context, rSpec *spec.Spec, options ..
 	}()
 
 	if rootless.IsRootless() && ctr.config.ConmonPidFile == "" {
-		ctr.config.ConmonPidFile = filepath.Join(ctr.state.RunDir, "conmon.pid")
+		ctr.config.ConmonPidFile = filepath.Join(ctr.config.StaticDir, "conmon.pid")
```

Comment:
Umm... if I am not mistaken, doesn't this mean that rootless containers created with earlier versions of `podman` can no longer be used with `podman exec`?

I wonder if we can soften the blow somehow. One option that came to my mind was to not hard depend on the `ConmonPidFile`. Maybe we can fallback to getting the PID of the container's PID 1 through `ctr.PID()`, and then walking back up the process tree to find the `conmon` process using some heuristics? Although, I don't know what could happen if someone tricks the heuristics into a false positive.

Reply:
I wonder if we can avoid writing the default `ConmonPidFile` path to the configuration when the user hadn't specified one explicitly? If we can do that, then it might open up some possibilities that aren't as extreme as "re-create the container and start afresh". :)

I don't know if it will work, but something like "you have to re-start the container" is a lot nicer.